component/org.wso2.carbon.identity.casque.authenticator/src/main/java/org/wso2/carbon/identity/casque/authenticator/authenticator/CasqueAuthenticator.java

```diff
@@ -0,0 +1,261 @@
+import org.wso2.carbon.identity.casque.authenticator.authenticator.internal.CasqueAuthenticatorServiceDataHolder;
+import org.wso2.carbon.identity.casque.authenticator.exception.CasqueException;
+import org.wso2.carbon.identity.casque.authenticator.authenticator.radius.Radius;
+import org.wso2.carbon.identity.casque.authenticator.authenticator.radius.RadiusResponse;
+
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * CASQUE Authenticator main class.
+ * Sends the user name and token Id to the CASQUE SNR Server using RADIUS protocol
+ * to get the challenge and also return the response.
+ * Authentication is a PASS for a valid response to the challenge.
+ */
+public class CasqueAuthenticator extends AbstractApplicationAuthenticator implements LocalApplicationAuthenticator {
+
+    private static final long serialVersionUID = 4341535155455223654L;
+    private static final Log log = LogFactory.getLog(CasqueAuthenticator.class);
+    private static final String CASQUE_SNR_CLAIM = "http://wso2.org/claims/identity/casqueSnrToken";
+    private static final String TOKEN_ID_FORMAT = "^[a-fA-F0-9]{3} [0-9]{6}$";  // e.g. "FFF 000001"
+    private static final AuthPages authPages = new AuthPages();
+
+    /**
+     * Get the users tokenId from the UserStoreManager.
+     * Check the format of the tokenId, 3 hex chars, a space then 6 numeric digits.
+     *
+     * @param userName Username of the user
+     * @return @return the CASQUE Token ID
+     * @throws CasqueException UserRealm for user or tenant domain is null
+     */
+    private String getCasqueTokenId(String userName) throws CasqueException {
+
+        try {
+            UserStoreManager userStoreManager = CasqueAuthenticatorServiceDataHolder.getInstance()
+                    .getRealmService().getTenantUserRealm(IdentityTenantUtil.getTenantIdOfUser(userName))
+                    .getUserStoreManager();
+
+            // Getting the Token ID assigned to userName
+            Map<String, String> tokenIdMap = userStoreManager.getUserClaimValues(userName,
+                    new String[]{CASQUE_SNR_CLAIM}, null);
+
+            if (tokenIdMap == null || tokenIdMap.get(CASQUE_SNR_CLAIM) == null) {
+                throw new CasqueException("Token ID is null for user: " + userName);
+            }
+            String tokenId = tokenIdMap.get(CASQUE_SNR_CLAIM);
+            if (tokenId.matches(TOKEN_ID_FORMAT)) {
+                return tokenId;
+            }
+            throw new CasqueException(String.format("%s is a bad formatted Token ID for user : %s", tokenId, userName));
+
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            log.error("User Store Exception:" + e.getMessage());
+        }
+        throw new CasqueException("Unable to get token id for user: " + userName);
+    }
+
+    /**
+     * Initiate the authentication request
```

Comment:
put full stop in after comments

Reply:
fixed in - https://github.com/wso2-extensions/identity-outbound-auth-casque-snr/pull/7